examples/src/index.tsx

```diff
@@ -1,49 +1,134 @@
 import queryString from 'query-string'
-import React from 'react'
+import * as React from 'react'
 import ReactDOM from 'react-dom'
 import Cropper from '../../src/index'
 import { Area, Point } from '../../src/types'
 import './styles.css'
 
+const TEST_IMAGES = {
+  '/images/dog.jpeg': 'Landscape',
+  '/images/flower.jpeg': 'Portrait',
+  '/images/cat.jpeg': 'Small portrait',
+  '/images/2000x1200.jpeg': '2000x1200',
+}
+
 const urlArgs = queryString.parse(window.location.search)
-const imageSrc = typeof urlArgs.img === 'string' ? urlArgs.img : '/images/dog.jpeg' // so we can change the image from our tests
+const imageSrcFromQuery =
+  typeof urlArgs.img === 'string' ? urlArgs.img : Object.keys(TEST_IMAGES)[0] // so we can change the image from our tests
+
+type HashType = 'percent' | 'pixel'
 
 type State = {
   imageSrc: string
   crop: Point
   rotation: number
   flip: { horizontal: boolean; vertical: boolean }
+  hashType: HashType
   zoom: number
   aspect: number
   cropShape: 'rect' | 'round'
   showGrid: boolean
   zoomSpeed: number
   restrictPosition: boolean
   croppedArea: Area | null
+  croppedAreaPixels: Area | null
+  initialCroppedAreaPercentages: Area | undefined
+  initialCroppedAreaPixels: Area | undefined
 }
 
+const hashNames = ['imageSrc', 'hashType', 'x', 'y', 'width', 'height', 'rotation'] as const
+
 class App extends React.Component<{}, State> {
-  state: State = {
-    imageSrc,
-    crop: { x: 0, y: 0 },
-    rotation: 0,
-    flip: { horizontal: false, vertical: false },
-    zoom: 1,
-    aspect: 4 / 3,
-    cropShape: 'rect',
-    showGrid: true,
-    zoomSpeed: 1,
-    restrictPosition: true,
-    croppedArea: null,
+  constructor(props: {}) {
+    super(props)
+
+    let rotation = 0
+    let initialCroppedAreaPercentages: Area | undefined = undefined
+    let initialCroppedAreaPixels: Area | undefined = undefined
+    let hashType: HashType = 'percent'
+    let imageSrc = imageSrcFromQuery
+
+    if (window && !urlArgs.setInitialCrop) {
```

Comment:
I think the hash part was useful while working on this PR but I'd like to keep this example simple. Do you agree to remove the hash-related logic from this file?
However please keep what you added for the image selection and the rotation ticks.

Reply:
I would argue that it's beneficial to keep the hash part because it helps to test the crop restoration easily - just refresh the page and check that everything stayed the same. For me it's not really an example, more like a development version for people who work on the lib. But up to you, I can remove it if you want

Reply:
Actually while testing in Firefox, I noticed that updating the URL on each event isn't accepted by it and crashes the example page 🙈
![image](https://user-images.githubusercontent.com/2678610/143492845-4a3fd73b-1bfd-4ec5-9211-5f99b52cdd90.png)

Reply:
I'll keep hash updates only for `onCropComplete` then, not for `onCropAreaChange`

Reply:
Actually, `onCropComplete` is called as well while zooming with the trackpad/mouse or rotating which also triggers the issue. Ideally, we should debounce in the example the logic to update the URL.

Having `onCropComplete` called this way is an issue in my opinion. We might want to debounce it in the future.

Reply:
Yes, you're right, i've added a debounce with 150ms, that seems to work well